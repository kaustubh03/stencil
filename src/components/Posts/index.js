// @flow
import React, { Component } from "react";
import { connect } from "react-redux";
import styles from "./index.module.css";
import { editPost } from "../../redux/actions/actionMain";
import { deletePost } from "../../redux/actions/actionMain";

class Posts extends Component {
  render() {
    return (
      <div className={styles.post}>
        <h2 className={styles.post_title}>
          {this.props.post ? this.props.post.title : null}
        </h2>
        <p className={styles.post_message}>
          {this.props.post ? this.props.post.message : null}
        </p>
        <div className={styles.control_buttons}>
          <button
            className={styles.edit}
            onClick={() =>
              this.props.dispatch(
                editPost(this.props.post ? this.props.post.id : null)
              )
            }
          >
            Edit
          </button>
          <button
            className={styles.delete}
            onClick={() =>
              this.props.dispatch(
                deletePost(this.props.post ? this.props.post.id : null)
              )
            }
          >
            Delete
          </button>
        </div>
      </div>
    );
  }
}
export default connect()(Posts);
