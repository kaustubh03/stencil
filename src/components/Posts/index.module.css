.Posts {
  display: block;
}
.post {
  background: #fff;
  width: 60%;
  margin: 0 auto;
  padding: 30px;
  margin-bottom: 20px;
  box-shadow: 10px 10px 12px 0 rgba(46, 60, 73, 0.05);
}
.post_title {
  text-align: center;
  font-weight: 400;
  font-size: 40px;
  color: #636363;
  word-wrap: break-word;
}

.post_message {
  font-size: 25px;
  font-weight: 200;
  line-height: 2.5rem;
  text-align: center;
  word-wrap: break-word;
}

.control_buttons {
  display: flex;
}

.delete {
  background: #ff7777;
  border: none;
  text-align: center;
  font-size: 20px;
  padding: 10px 25px;
  border-radius: 20px;
  cursor: pointer;
  outline: none;
  color: #fff;
}
.edit {
  background: #02b3e4;
  border: none;
  text-align: center;
  font-size: 20px;
  padding: 10px 25px;
  border-radius: 20px;
  cursor: pointer;
  margin-right: auto;
  outline: none;
  color: #fff;
}
