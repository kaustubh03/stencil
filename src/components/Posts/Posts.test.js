import React from "react";
import renderer from "react-test-renderer";
import Posts from ".";
import configureStore from "../../redux/configureStore";
const store = configureStore();

it("Posts: default", () => {
  const component = renderer.create(<Posts store={store} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
