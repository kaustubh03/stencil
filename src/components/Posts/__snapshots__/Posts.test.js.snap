// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts: default 1`] = `
<div
  className="post"
>
  <h2
    className="post_title"
  />
  <p
    className="post_message"
  />
  <div
    className="control_buttons"
  >
    <button
      className="edit"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
